### DISCUSSION TITLE
Feedback for If Statements lesson

### DISCUSSION THREAD

**Discussion by jcollard** [CaptainCoderOrg/IntroToCSharpSite#143] [Page Feedback]
Feedback for If Statements lesson: https://csharp-develop.captaincoder.org/lessons/branching-logic/if-statements

**Comment by cloudthepain** [comment]
In your adventure text you make the statement that they already completed the Shape Analyzer and then in the next slide recommend that they try it. I would look into adjust it.
Why is the Character Creator a great place to level up their skills?

(What do you think about this re-write? Currently it is a little wordy.)
"Imagine you are creating an adventure game and..." you want to give the player the option to choose where they go. One way you could do this is by giving them a list of options to choose from in a menu. Each option would need to provide a different outcome depending on the choice that they made.

Note: "... conditional or the condition of the if statement." This is because depending on the condition, a certain result may or may not take place.


"When the **condition** (bool expression) of an if statement is true, the body of the if statement executes." (Do you think this reads better?)

Do you think that the operators should be split in another section?
I think that nested if statements should be moved to the bottom.

"Often times, you would like to execute a different body when the condition is false." Sometimes, however, we want our program to perform an action if the **if statement** isn't performed. In this case we would use the **else statement.**


The Recipe for an if statement is

// Do this if the **if condition is true**

//Otherwise do this.



"When the program reaches the if statement it is going to choose to execute exactly one "branch" of code." That means that when you use an if-else statement, the program **must ** pick one of the choices you give it.

The Recipe for an else if statement is

// Do this if the **if condition is true**

//Or **if this condition is true** do this.

**Note** Unlike an if statement followed by an **else** statement, an **if-else** statement doesn't always have to execute one of the branches in its tree. That means that the both the if statement and the else-if statement could be ignored if you wanted to.
(Add example where both are ignored.)


We could add a picture of a decision tree to help make this concept more clear.


It would be nice at the end of the step-by-step example if we could jump back to where we entered the number.


"You may have noticed that the programs you've written so far require(**s**) your user to enter..."

**Reply by jcollard** [reply]
`Do you think that the operators should be split in another section?`

Yes. But, I think it is also okay to include `==` and `!=` here as a little taste. There will be another section called "Comparison Operators" or maybe (relational operators).

**Reply by jcollard** [reply]
```
"When the program reaches the if statement it is going to choose to 
execute exactly one "branch" of code." That means that when you use 
an if-else statement, the program **must ** pick one of the choices you 
give it.
```

What do you think of:

```
Note: When you add an else statement to the end of an if statement 
you create a decision tree with two branches. As the computer executes 
the if statement it chooses a branch based on the condition of the if 
statement. This means that when you use an if-else statement, the 
program MUST pick exactly one of the two branches.
```

**Reply by jcollard** [reply]
```
Note Unlike an if statement followed by an else statement, an if-else statement 
doesn't always have to execute one of the branches in its tree. That means that the 
both the if statement and the else-if statement could be ignored if you wanted to.
(Add example where both are ignored.)
```

What about:
```
Note: Similar to an else statement, adding an else if statement to the end of an if 
statement creates a decision tree with two branches. However, when the condition 
of the first branch is false, the computer does not automatically execute the body 
of the second branch. Instead, it must first check the condition of the second 
branch. If the condition is true the body is executed. If the condition is false the 
body is skipped. This means if both conditions evaluate to false, the computer will 
not execute either branch.
```

**Reply by cloudthepain** [reply]
> Note: When you add an else statement to the end of an if statement 
> you create a decision tree with two branches. As the computer executes 
> the if statement it chooses **one of the branches** based on the condition of the if 
> statement. This means that when you use an if-else statement, the 
> program MUST pick exactly one of the two branches.

------------------------------------------------------------------
Typo:
 If the condition is true**,** the body is executed.

-------------

I think that your adjustments are an improvement :D

**Comment by jcollard** [comment]
Thanks for the feedback! I've incorporated it and redeployed.
